crates/pallets/starknet/src/transaction_validation.rs

```diff
@@ -50,3 +53,87 @@ impl<OuterOrigin: Into<Result<RawOrigin, OuterOrigin>> + From<RawOrigin>> Ensure
+                    None => return Ok(TxPriorityInfo::InvokeV0),
+                };
+
+                // Reject transaction with an already used Nonce
+                if sender_nonce > transaction_nonce {
+                    Err(InvalidTransaction::Stale)?;
+                }
+
+                // A transaction with a nonce higher than the expected nonce is placed in
+                // the future queue of the transaction pool.
+                if sender_nonce < transaction_nonce {
+                    log::debug!(
+                        "Nonce is too high. Expected: {:?}, got: {:?}. This transaction will be placed in the \
+                         transaction pool and executed in the future when the nonce is reached.",
+                        sender_nonce,
+                        transaction_nonce
+                    );
+                }
+
+                Ok(TxPriorityInfo::RegularTxs { sender_address: tx.sender_address(), transaction_nonce, sender_nonce })
+            }
+            UserAndL1HandlerTransaction::L1Handler(tx, _fee) => {
+                Self::ensure_l1_message_not_executed(&Nonce(StarkFelt::from(tx.nonce)))?;
+                Ok(TxPriorityInfo::L1Handler)
+            }
+        }
+    }
+
+    pub fn validate_unsigned_tx(transaction: &UserAndL1HandlerTransaction) -> Result<(), InvalidTransaction> {
+        let chain_id = Self::chain_id();
+        let block_context = Self::get_block_context();
+        let mut state: BlockifierStateAdapter<T> = BlockifierStateAdapter::<T>::default();
+        let mut execution_resources = ExecutionResources::default();
+        let mut initial_gas = blockifier::abi::constants::INITIAL_GAS_COST;
+
+        match transaction {
+            UserAndL1HandlerTransaction::User(transaction) => {
+                match transaction {
+                    // There is no way to validate it before the account is actuallly deployed
+                    UserTransaction::DeployAccount(_) => Ok(None),
+                    UserTransaction::Declare(tx, contract_class) => tx
+                        .try_into_executable::<T::SystemHash>(chain_id, contract_class.clone(), false)
+                        .map_err(|_| InvalidTransaction::BadProof)?
+                        .validate_tx(&mut state, &block_context, &mut execution_resources, &mut initial_gas, false),
+                    UserTransaction::Invoke(tx) => tx.into_executable::<T::SystemHash>(chain_id, false).validate_tx(
+                        &mut state,
+                        &block_context,
+                        &mut execution_resources,
+                        &mut initial_gas,
+                        false,
+                    ),
+                }
+            }
+            UserAndL1HandlerTransaction::L1Handler(transaction, fee) => transaction
+                .into_executable::<T::SystemHash>(chain_id, *fee, false)
+                .validate_tx(&mut state, &block_context, &mut execution_resources, &mut initial_gas, false),
```

Comment:
What are we exactly validating here? I guess this calls the`__validate__` function.

Reply:
Yes exactly. The one of the account being targeted by this message

Reply:
But can't l1 messages be sent to smart contracts directly? For example, when you bridge from ETH, the call goes directly to the ETH smart contract on Starknet to mint the tokens. You don't have an account on the L2 at that time which would validate.

Example tx - https://voyager.online/tx/0x62822a36c3d819bae23347ec6df4b5a46c2cef939175f842ed1b7ac7004c009

Reply:
I think you are write, we were tripping and there is just no validation (other than the nonce one) for L1Handler tx